torch/lib/THD/base/init_methods/InitMethod.hpp

```diff
@@ -11,23 +11,56 @@ namespace thd {
 struct InitMethod {
   struct Config {
     struct MasterConfig {
-      rank_type world_size;
       int listen_socket;
       port_type listen_port;
     };
 
     struct WorkerConfig {
-      std::string address;
-      port_type port;
+      std::string master_addr;
+      port_type master_port;
     };
 
+    Config() {
+      rank = INT32_MAX - 1;
+      world_size = 0;
+      public_address = "";
+      master.listen_socket = 0;
+      master.listen_port = 0;
+      worker.master_addr = "";
+      worker.master_port = 0;
+    }
+
     rank_type rank;
+    rank_type world_size;
     std::string public_address;
     MasterConfig master;
     WorkerConfig worker;
-  };
 
-  virtual Config getConfig() = 0;
+    void validate() {
```

Comment:
It just checks if config is filled (we do not care how). Other checking should be done inside particular init methods.